### PR TITLE
[TASK] Config: Add MySQL environment variables

### PR DESCRIPTION
**Pull request by M-arcus**
See https://github.com/friendica/friendica/issues/4791

### CODE REVIEW

**Comment by M-arcus** [comment]
My use case is that I have a docker container and it changes the mysql config after every start, but allows the PHP to get the config from the environment variables.

**Comment by MrPetovan** [comment]
Can you make the environment variables more specific and check for their value to decide if they should be pre-populated or not?

**Comment by M-arcus** [comment]
If you mean changing their name i.e. `MYSQL_USERNAME` to something else, no. The mysql docker file i have to use needs that variable-name.

**Comment by MrPetovan** [comment]
Can you change the MySQL Docker file? It may not be an issue, you can still check that all environment variables have a value before overriding the expected values.

**Comment by M-arcus** [comment]
I can't change that one either, sorry.

**Comment by MrPetovan** [comment]
Okay, just test for all environment variables values before overriding the existing ones. No need for manual edit of the file.

File: htconfig.php [hunk 1]
```diff
@@ -21,6 +21,15 @@
 $db_pass = 'mysqlpassword';
 $db_data = 'mysqldatabasename';
 
+// Set this variable to true if you want to use environment variables for mysql
+$db_use_env_vars = false; 
+if ($db_use_env_vars) {
```
Comment (MrPetovan) [inline htconfig.php]: Pleas test for environment variables values before overriding the existing ones instead of a manual switch.

Comment (M-arcus) [reply htconfig.php]: Should i only test for one, or do a check for all of them?

File: view/templates/htconfig.tpl [hunk 2]
```diff
@@ -17,6 +17,15 @@ $db_user = '{{$dbuser}}';
 $db_pass = '{{$dbpass}}';
 $db_data = '{{$dbdata}}';
 
+// Set this variable to true, if you want to use environment variables for mysql
+$db_use_env_vars = false;
+if ($db_use_env_vars === true) {
```
[the remark this answers is not in the document]
Comment (M-arcus) [reply view/templates/htconfig.tpl]: Added.

**Comment by MrPetovan** [comment]
Question: if the default MySQL port is used, is the environment variable MYSQL_PORT set t the default value as well?

**Comment by M-arcus** [comment]
@MrPetovan Yes, the port is set, even when it's the default.
Should i make it optional?

**Comment by MrPetovan** [comment]
No, I was just wondering whether the environment variable was set even if no custom port was specified. Thanks!

**Comment by MrPetovan** [comment]
Documentation is needed now but it can be done in a separate PR.

**Pull request merged**